OutOfSchool/OutOfSchool.WebApi/Services/StatisticService.cs

```diff
@@ -57,115 +62,141 @@ public async Task<IEnumerable<DirectionStatistic>> GetPopularDirections(int limi
+                        .Where(w => string.Equals(w.Address.City, city.Trim()));
+                }
 
-            // LEFT JOIN DirectionsWithWorkshops with DirectionsWithApplications
-            var directionsWithCounts = directionsWithWorkshops
-                .GroupJoin(
-                    directionsWithApplications,
-                    directionWithWorkshop => directionWithWorkshop.DirectionId,
-                    directionWithApplication => directionWithApplication.DirectionId,
-                    (directionWithWorkshop, localDirectionsWithApplications) => new
-                    {
-                        directionWithWorkshop,
-                        localDirectionsWithApplications,
-                    })
-                .SelectMany(
-                    x => x.localDirectionsWithApplications.DefaultIfEmpty(),
-                    (x, y) => new
+                var directionsWithWorkshops = workshops
+                    .GroupBy(w => w.DirectionId)
+                    .Select(g => new
                     {
-                        DirectionId = x.directionWithWorkshop.DirectionId,
-                        ApplicationsCount = y.ApplicationsCount,
-                        WorkshopsCount = x.directionWithWorkshop.WorkshopsCount,
+                        DirectionId = g.Key,
+                        WorkshopsCount = g.Count() as int?,
                     });
 
-            var allDirections = directionRepository.Get<int>();
-
-            // LEFT JOIN DirectionsWithCounts with all Directions
-            var statistics = allDirections
-                .GroupJoin(
-                    directionsWithCounts,
-                    direction => direction.Id,
-                    directionWithCounts => directionWithCounts.DirectionId,
-                    (direction, localDirectionsWithCounts) => new { direction, localDirectionsWithCounts })
-                .SelectMany(
-                    x => x.localDirectionsWithCounts,
-                    (x, y) => new DirectionStatistic
+
+                var directionsWithApplications = applications
+                    .GroupBy(a => a.Workshop.DirectionId)
+                    .Select(g => new
                     {
-                        Direction = x.direction.ToModel(),
-                        ApplicationsCount = y.ApplicationsCount ?? 0,
-                        WorkshopsCount = y.WorkshopsCount ?? 0,
+                        DirectionId = g.Key,
+                        ApplicationsCount = g.Count() as int?,
                     });
 
-            var sortedStatistics = await statistics
-                .OrderByDescending(s => s.ApplicationsCount)
-                .Take(limit)
-                .ToListAsync()
-                .ConfigureAwait(false);
-
-            logger.LogInformation($"All {sortedStatistics.Count} records were successfully received");
-
-            return sortedStatistics;
+                // LEFT JOIN DirectionsWithWorkshops with DirectionsWithApplications
```

Comment:
Your comment states that it is `LEFT JOIN` (aka `LEFT OUTER JOIN`), but I'm afraid that `GroupJoin` is `INNER JOIN`. 
As the comment doesn't match the code, it's unclear - is the comment wrong or the code is wrong

Reply:
This part of code was written by other people. We can errase this comment :)

Reply:
Comments have been deleted )